server/datasources/cachedMongooseDataSource.ts

```diff
@@ -10,6 +10,27 @@ export default class CachedMongooseDataSource<DocumentType extends Document> ext
       this.cache = cache
   }
 
+  async softDeleteRequestType(id: Types.ObjectId) {
```

Comment:
Let's make the function name and variable names generic. 
Replace `softDeleteRequestType` with `softDeleteById` (`resolvers.ts` will need to be changed too)
Directly use `this.cache.model` instead of `RequestType` on line 15

```
async softDeleteById(id: Types.ObjectId) {
  const promise = this.cache.model.findByIdAndUpdate(id, {"deleted": true})
    .then(res => {
      return {
        "success": true,
        "message": `${this.cache.name} successfully deleted`,
        "id": id
      }
    })
    .catch(error => {
      console.log(error)
      return {
        "success": false,
        "message": error._message,
        "id": id
      }
    })
  return promise
}
```